skbeam/core/speckle.py

```diff
@@ -237,20 +240,29 @@ def _process(num_roi, level, buf_no, buf, img_per_level, labels,
         probability density of detecting photons
     prob_k_pow : array
         squares of probability density of detecting photons
+    track_bad : array
+        to track bad images in each level
     """
     img_per_level[level] += 1
     u_labels = list(np.unique(labels))
 
-    for j, label in enumerate(u_labels):
-        roi_data = buf[level, buf_no][labels == label]
-        spe_hist, bin_edges = np.histogram(roi_data, bins=bin_edges,
-                                           density=True)
-        spe_hist = np.nan_to_num(spe_hist)
-        prob_k[level, j] += (spe_hist -
-                             prob_k[level, j])/(img_per_level[level])
-
-        prob_k_pow[level, j] += (np.power(spe_hist, 2) -
-                                 prob_k_pow[level, j])/(img_per_level[level])
+    # check if there are any bad images, using bad_to_nan function
```

Comment:
Thank you for the clarification.  I misunderstood the intent of the code comment because I interpreted this code comment as "This code is going to check for bad images using the bad_to_nan function".  Would you change this code comment to simply say "Check if there are any bad images, represented as an array filled with np.nan"?  I think that would capture your intent.